inference-engine/src/mkldnn_plugin/nodes/extract_image_patches.cpp

```diff
@@ -0,0 +1,242 @@
+// Copyright (C) 2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "list.hpp"
+#include "base.hpp"
+#include "details/caseless.hpp"
+
+#include <cmath>
+#include <string>
+#include <vector>
+#include <cassert>
+#include <set>
+#include "ie_parallel.hpp"
+
+namespace InferenceEngine {
+namespace Extensions {
+namespace Cpu {
+
+class ExtractImagePatchesImpl : public ExtLayerBase {
+public:
+    explicit ExtractImagePatchesImpl(const CNNLayer* layer) {
+        try {
+            if (details::CaselessEq<std::string>()("ExtractImagePatchesLayer", layer->type))
+                THROW_IE_EXCEPTION << "Layer " << layer->type << " with name '" << layer->name
+                    << "' is not instance of ExtractImagePatchesLayer class";
+
+            if (layer->insData.size() != 1 || layer->outData.size() != 1)
+                THROW_IE_EXCEPTION << "Layer " << layer->type << " with name '" << layer->name
+                    << "' has incorrect number of input or output edges!";
+
+            auto inData = layer->insData[0].lock();
+            if (inData == nullptr)
+                THROW_IE_EXCEPTION << "Layer " << layer->type << " with name '" << layer->name
+                    << "' has nullable input data";
+
+            if (inData->getLayout() != NCHW)
+                THROW_IE_EXCEPTION << "Layer " << layer->type << " with name '" << layer->name
+                    << "' has unsupported layout: " << inData->getLayout();
+
+            const auto precision = inData->getTensorDesc().getPrecision();
+            if (_supported_precisions_sizes.find(precision.size()) == _supported_precisions_sizes.end())
+                THROW_IE_EXCEPTION <<  "Layer " << layer->type << " with name '" << layer->name
+                    << "' has unsupported precision: " << precision.name();
+
+            auto ksizes = layer->GetParamAsUInts("sizes", {1u, 1u});
+            auto strides = layer->GetParamAsUInts("strides", {1u, 1u});
+            auto rates = layer->GetParamAsUInts("rates", {1u, 1u});
+            _padding = layer->GetParamAsString("auto_pad", "valid");
```

Comment:
Please also check that auto_pad parameter contains supported value. W/o this check the implementation will became invalid in case new possible value will be added for "auto_pad" inside ngraph.

Reply:
Done